source/agora/test/Base.d

```diff
@@ -1363,14 +1420,20 @@ public enum NetworkTopology
 /// Node / Network / Quorum configuration for use with makeTestNetwork
 public struct TestConf
 {
+    /// Verbose test logging
+    bool verbose_test_logging = false;
+
     /// Network topology to use
     NetworkTopology topology = NetworkTopology.FullyConnected;
 
     /// Extra blocks to generate in addition to the genesis block
     size_t extra_blocks = 0;
 
-    /// Number of validator nodes to instantiate
-    size_t validators = 4;
+    /// Number of validator nodes that are to be enrolled in test setup
+    size_t validators = 6;
+
+    /// Number of validator nodes that are enrolled in test `GenesisBlock`
+    size_t genesis_validators = 6;
```

Comment:
This should be implicit, not a config option.

Reply:
Actually I don't understand why now there are 3 options: `validators`, `genesis_validators`, and `outsider_validators`. Surely there should be just one option: `outsider_validators`? Genesis should contain only 6, and the outsiders are customizable.

Reply:
The idea I was having is to remove the outsider_validators after the tests are refactored. But you are right that the genesis_validators should be hard coded.